models/Article.php

```diff
@@ -106,108 +95,218 @@ function get_post(){
+                $errors['empty'] = "Veuillez remplir tous les champs";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable";
+                }
+            }
+
+            if(!empty($errors)){
+                ?>
+                    <div class="card red">
+                        <div class="card-content white-text">
+                            <?php
+                                foreach($errors as $error){
+                                    echo $error;
+                                }
+                            ?>
+                        </div>
+                    </div>
+                <?php
+            }else{
+                Article::post($title,$content,$posted);
+                if(!empty($_FILES['image']['name'])){
+                    Article::post_img($_FILES['image']['tmp_name'], $extension);
+                    header("Location:/liste-de-tous-les-articles");
+                }else{
+                    $id = $this->db->lastInsertId();
+                   header("Location:/liste-de-tous-les-articles");
+                }
+            }
+        }
+    }
+    
+    function form_page_postback(){
+        if(isset($_POST['delete'])){
+            Article::delete_article();
+            Article::delete_article_comments();
         }
 
-        $query = $db->prepare('DELETE FROM articles WHERE id = :id');
-        $query->execute(['id' => $id]);
+        if(isset($_POST['submit'])){
+            if(isset($_POST['title'])){
+                $title = htmlspecialchars(trim($_POST['title']));
+            }
+            if(isset($_POST['content'])){
+                $content = htmlspecialchars(trim($_POST['content']));
+            }
+            $posted = isset($_POST['public']) ? "1" : "0";
+            
+            $errors = [];
 
-        exit();
+            if(empty($title) || empty($content)){
+                $errors['empty'] = "Veuillez remplir tous les champs svp";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+  
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable.";
+                }
+            }
+            
+            if(!empty($errors)){
+                ?>
+                <div class="card red">
+                    <div class="card-content white-text">
+                        <?php
+                        foreach($errors as $error){
+                            echo $error;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$error'.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5344015)